modules/custom/az_publication/az_publication.module

```diff
@@ -306,12 +352,22 @@ function az_publication_node_view(array &$build, EntityInterface $entity, Entity
           "author" => $renderlink,
           "csl-entry" => function ($cslItem, $renderedText) {
             // Remove citation number tag.
-            // In general it's never a good idea to parse html with regex.
-            // This works ONLY because the output is a known pattern.
             if (!empty($cslItem->additionalLinks)) {
               $renderedText .= ' ' . $cslItem->additionalLinks;
             }
-            return preg_replace('#<div class="csl-left-margin">(.*?)</div>#', '', $renderedText);
+            $dom = new DOMDocument();
+            // Suppress loadHTML warnings.
+            libxml_use_internal_errors(TRUE);
+            $dom->loadHTML($renderedText);
+            libxml_clear_errors();
+            $divs = $dom->getElementsByTagName('div');
+            foreach ($divs as $div) {
+              if ($div->getAttribute('class') === 'csl-left-margin') {
+                $div->parentNode->removeChild($div);
+              }
+            }
+            $renderedText = $dom->saveHTML();
+            return $renderedText;
```

Comment:
This change breaks multibyte characters:
```
Wildcat, Wilbur. â€œTest Publicationâ€. Journal of Wildcats, vol. IV, no. 17, Nov. 2022, pp. 1-50.
```

```suggestion
            $dom = new DOMDocument();
            // Suppress loadHTML warnings.
            libxml_use_internal_errors(TRUE);
            $dom->loadHTML('<?xml encoding="UTF-8">' . $renderedText);
            libxml_clear_errors();
            $divs = $dom->getElementsByTagName('div');
            foreach ($divs as $div) {
              if ($div->getAttribute('class') === 'csl-left-margin') {
                $div->parentNode->removeChild($div);
              }
            }
            $renderedText = $dom->saveHTML();
            return $renderedText;
```

or usage of `mb_convert_encoding`?